app/models/concerns/unparsed_run.rb

```diff
@@ -13,154 +18,286 @@ def parse_into_db
-
-          gametime_skipped: parsed_segment[:gametime_skipped],
-          gametime_reduced: false,
-          gametime_gold:    parsed_segment[:gametime_gold]
+          segment_number: i,
+          name:           lsc_segment.name || '',
+          icon:           lsc_segment.icon.presence # do something with this
         )
+
+        %w[real game].each do |timing|
+          segment.send(
+            "#{timing}time_end_ms=",
+            lsc_segment.personal_best_split_time.send("#{timing}_time")&.total_seconds&.*(1000) || 0
+          )
+          segment.send(
+            "#{timing}time_duration_ms=",
+            segment.send("#{timing}time_end_ms") - previous_segment.send("#{timing}time_end_ms")
+          )
+          segment.send(
+            "#{timing}time_start_ms=",
+            segment.send("#{timing}time_end_ms") - segment.send("#{timing}time_duration_ms")
+          )
+          segment.send(
+            "#{timing}time_shortest_duration_ms=",
+            lsc_segment.best_segment_time.send("#{timing}_time")&.total_seconds&.*(1000) || 0
+          )
+          segment.send(
+            "#{timing}time_skipped=",
+            segment.send("#{timing}time_duration_ms").zero?
+          )
+          segment.send(
+            "#{timing}time_gold=",
+            segment.send("#{timing}time_duration_ms") <= segment.send("#{timing}time_shortest_duration_ms")
+          )
+          segment.send(
+            "#{timing}time_reduced=",
+            false
+          )
+        end
+
+        container[lsc_segment] = segment
+        previous_segment = segment
+        next if container.size < MAX_CONTAINER_SIZE
+
+        result = write_segments(container.values)
+        process_segment_histories(container.keys, result.ids)
+        container = {}
       end
 
-      segments.import(
-        %i[
-          run_id
-          segment_number
-          name
-
-          realtime_start_ms
-          realtime_end_ms
-          realtime_duration_ms
-          realtime_shortest_duration_ms
-          realtime_skipped
-          realtime_reduced
-          realtime_gold
-
-          gametime_start_ms
-          gametime_end_ms
-          gametime_duration_ms
-          gametime_shortest_duration_ms
-          gametime_skipped
-          gametime_reduced
-          gametime_gold
-        ],
-        segs
-      )
-
-      segs
+      return if container.blank?
+
+      result = write_segments(container.values)
+      process_segment_histories(container.keys, result.ids)
     end
 
-    def write_segment_histories(segs)
-      ids = segments.order(:segment_number).pluck(:id)
-      histories = []
+    def write_segments(segments)
+      segments.each do |seg|
+        %w[real game].each do |timing|
+          %w[time_end_ms time_duration_ms time_shortest_duration_ms].each do |field|
+            value = seg.send("#{timing}#{field}")
+            seg.send("#{timing}#{field}=", zero_to_nil(value))
```

Comment:
I needed the value to be zero for the durations in the original loop because I calculate time a little differently, which is why we `zero_to_nil` them here instead.  The new calculation should stop runs from being seen as done if any split has a time (which I believe happens when you reset on your first run, I still need to test this one).